org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TreeViewerDoubleClickListener.java

```diff
@@ -13,8 +13,11 @@
 
 import com.google.common.base.Preconditions;
 
```

Comment:
This code is missing test coverage. We should have functional tests that
1. imports a project
2. double-clicks on collapsed and expanded project-group-task nodes
3. verifies if the proper expand/collapse/execution action happened.

Please have a look at the `TaskExecutionTest` class, you should be able to create a test based on that one. There are some rough edges there so don't hesitate to ask if something is not clear.